### PR TITLE
Update helper to better handle Xcode Server

### LINKED ISSUE #11605: Pilot fails to upload ipa: itunes_transporter can't initialise
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I use fastlane to upload the ipa to the store in a script run by an Xcode bot.
The system is a clean install of 10.13 with the newest Xcode 9.2.

The command I run is:
`fastlane pilot upload --skip_submission --skip_waiting_for_build_processing -i "myApp.ipa" -u "my@account.com"`

#### Result
```
[✔] 🚀
[11:17:40]: Login to iTunes Connect
[11:17:44]: Login successful
[11:17:45]: Ready to upload new build to TestFlight ...
[11:17:47]: Sending crash report...

Looking for related GitHub issues on fastlane/fastlane...

Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
/Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:310:in `initialize': [!] undefined method `start_with?' for nil:NilClass (NoMethodError)
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/pilot/lib/pilot/build_manager.rb:28:in `new'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/pilot/lib/pilot/build_manager.rb:28:in `upload'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/pilot/lib/pilot/commands_generator.rb:66:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/pilot/lib/pilot/commands_generator.rb:166:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/pilot/lib/pilot/commands_generator.rb:18:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/fastlane/lib/fastlane/cli_tools_distributor.rb:86:in `take_off'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.76.1/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:22:in `load'
	from /usr/local/bin/fastlane:22:in `<main>'
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.13.2                                                        |
| Ruby                        | 2.3.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.14.3 (Apple Git-98)                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.13.2 (17C205)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.2.7                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Library/Developer/CommandLineTools/                           |
| Xcode Version               | 9.2                                                               |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_GB.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.76.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 2.3.2        |
| CFPropertyList            | 2.3.6        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.60.0       |
| faraday                   | 0.13.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| io-console                | 0.4.5        |
| libxml-ruby               | 2.9.0        |
</details>

*generated on:* **2018-01-18**
</details>

### PR DESCRIPTION
**Pull request by taquitos**
This should help identify and/or fix #11605

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 1]
```diff
@@ -152,17 +152,36 @@ def self.buildlog_path
...
+        # Xcode server always creates a link here
+        return "/Library/Developer/XcodeServer/CurrentXcodeSymlink/Contents/Developer"
+      end
+
+      return `xcode-select -p`.delete("\n") + "/"
+    end
+
+    def self.xcode_server?
+      # XCS is set by Xcode Server
+      is_xcode_server = ENV["XCS"] || 0
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/helper.rb]: This whole method could also be `ENV["XCS"].to_i == 1`, right?

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 2]
```diff
@@ -152,17 +152,36 @@ def self.buildlog_path
     #  running system
     def self.xcode_path
       return "" unless self.is_mac?
-      `xcode-select -p`.delete("\n") + "/"
+
+      if self.xcode_server?
+        # Xcode server always creates a link here
+        return "/Library/Developer/XcodeServer/CurrentXcodeSymlink/Contents/Developer"
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/helper.rb]: To what version of Xcode does this link to then? Maybe we should add a `UI.verbose` here to log that we're doing this

**Review by KrauseFx — APPROVED** [review]
Legit 👍

File: fastlane_core/lib/fastlane_core/helper.rb (hunk 1 above)
Comment (taquitos) [reply fastlane_core/lib/fastlane_core/helper.rb]: yup

File: fastlane_core/lib/fastlane_core/helper.rb (hunk 2 above)
Comment (taquitos) [reply fastlane_core/lib/fastlane_core/helper.rb]: This links to whatever XcodeServer has determined it will use.

**Pull request merged**